tfhe/Cargo.toml

```diff
@@ -78,29 +82,49 @@ getrandom = { version = "0.2.8", optional = true }
 bytemuck = "1.13.1"
 
 [features]
+forward_compatibility = [
+    "dep:bincode",
+    "dep:tfhe-c-api-dynamic-buffer",
+    "dep:next_tfhe",
+    "dep:paste",
+]
+
 # paste is used by the HL API
-boolean = ["dep:paste"]
-shortint = ["dep:paste"]
-integer = ["shortint", "dep:paste"]
-internal-keycache = ["lazy_static", "dep:fs2", "dep:bincode", "dep:paste"]
+boolean = ["dep:paste", "next_tfhe?/boolean"]
+shortint = ["dep:paste", "next_tfhe?/shortint"]
+integer = ["shortint", "dep:paste", "next_tfhe?/integer"]
+internal-keycache = [
+    "dep:lazy_static",
+    "dep:fs2",
+    "dep:bincode",
+    "dep:paste",
+    "next_tfhe?/internal-keycache",
+]
 safe-deserialization = ["dep:bincode"]
 
 # Experimental section
-experimental = []
-experimental-force_fft_algo_dif4 = []
+experimental = ["next_tfhe?/experimental"]
+experimental-force_fft_algo_dif4 = [
+    "next_tfhe?/experimental-force_fft_algo_dif4",
+]
 # End experimental section
 
-__c_api = ["cbindgen", "dep:bincode", "dep:paste"]
+__c_api = [
+    "dep:cbindgen",
+    "dep:bincode",
+    "dep:paste",
+    "next_tfhe?/__force_skip_cbindgen",
+]
 boolean-c-api = ["boolean", "__c_api"]
 shortint-c-api = ["shortint", "__c_api"]
 high-level-c-api = ["boolean-c-api", "shortint-c-api", "integer", "__c_api"]
 
 __wasm_api = [
-    "wasm-bindgen",
-    "js-sys",
-    "console_error_panic_hook",
-    "serde-wasm-bindgen",
-    "getrandom",
+    "dep:wasm-bindgen",
+    "dep:js-sys",
+    "dep:console_error_panic_hook",
+    "dep:serde-wasm-bindgen",
+    "dep:getrandom",
```

Comment:
I don't understand why there are dependencies with `dep:` prefix and others not.
Can I have the rationale behind or link toward this feature ?

Reply:
https://doc.rust-lang.org/cargo/reference/features.html#optional-dependencies

Reply:
so normally all deps are now prefixed with dep: and features are not prefixed with dep:

it's something new-ish from cargo to avoid collisions between dependencies and features I believe